### PR TITLE
dragonbox: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **dragonbox/1.1.3**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dragonbox/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
```
Comment (uilianries) [inline recipes/dragonbox/all/conanfile.py:50]: It's a C++ project, but no compiler configurations? 🤔

Comment (toge) [reply recipes/dragonbox/all/conanfile.py:50]: You are right.
It's my mistake.

File: recipes/dragonbox/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+sources:
+  "1.1.3":
+    url: "https://github.com/jk-jeon/dragonbox/archive/refs/tags/1.1.3.tar.gz"
+    sha256: "09d63b05e9c594ec423778ab59b7a5aa1d76fdd71d25c7048b0258c4ec9c3384"
+patches:
+  "1.1.3":
+    - patch_file: "patches/1.1.3-fix-include-directory.patch"
+      patch_description: "fix include path"
+      patch_type: "portability"
```
Comment (uilianries) [inline recipes/dragonbox/all/conandata.yml:9]:
```suggestion
      patch_type: "conan"
```

https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#portability

I don't see a new support, but a change to make it work with Conan.

File: recipes/dragonbox/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class DragonboxConan(ConanFile):
+    name = "dragonbox"
+    description = "Reference implementation of Dragonbox in C++"
+    license = "Apache-2.0", "BSL-1.0"
```
Comment (uilianries) [inline recipes/dragonbox/all/conanfile.py:18]:
```suggestion
    license = ("Apache-2.0-with", "LLVM-exception", "BSL-1.0")
```
The LLVM exception is listed in Apache license file: https://github.com/jk-jeon/dragonbox/blob/master/LICENSE-Apache2-LLVM

According to the SPDX page, there is an official identifier for LLVM exception: https://spdx.org/licenses/LLVM-exception.html

Comment (toge) [reply recipes/dragonbox/all/conanfile.py:18]: Thanks!

File: recipes/dragonbox/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class DragonboxConan(ConanFile):
+    name = "dragonbox"
+    description = "Reference implementation of Dragonbox in C++"
+    license = ("Apache-2.0-with", "LLVM-exception", "BSL-1.0")
```
Comment (uilianries) [inline recipes/dragonbox/all/conanfile.py:18]:
```suggestion
    license = ("Apache-2.0", "LLVM-exception", "BSL-1.0")
```
Ops!

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**